deepchecks/nlp/context.py

```diff
@@ -72,68 +73,85 @@ class _DummyModel(BasicModel):
                 # # TODO: Uncomment when text data indices are not len(range(len(data)))
                 # get_logger().warning('train and test datasets have common index - adding "train"/"test"'
                 #                      ' prefixes. To avoid that provide datasets with no common indexes '
                 #                      'or pass the model object instead of the predictions.')
 
-        for dataset, y_pred, y_proba in zip([train, test],
-                                            [y_pred_train, y_pred_test],
-                                            [y_proba_train, y_proba_test]):
-            if dataset is not None:
-                if y_pred is not None:
-                    self._validate_prediction(dataset, y_pred, len(model_classes))
-                if y_proba is not None:
-                    self._validate_proba(dataset, y_proba, len(model_classes))
-
-                if dataset.task_type == TaskType.TEXT_CLASSIFICATION:
-                    if (y_pred is None) and (y_proba is not None):
-                        if dataset.is_multi_label_classification():
-                            y_pred = (np.array(y_proba) > 0.5)  # TODO: Replace with user-configurable threshold
-                        else:
-                            y_pred = np.argmax(np.array(y_proba), axis=-1)
-                            y_pred = np.array(model_classes, dtype='str')[y_pred]
-
-                    if y_pred is not None:
-                        if dataset.is_multi_label_classification():
-                            y_pred = np.array(y_pred)
-                        else:
-                            y_pred = np.array(y_pred, dtype='str')
-                        if len(y_pred.shape) > 1 and y_pred.shape[1] == 1:
-                            y_pred = y_pred[:, 0]
-                        ensure_predictions_shape(y_pred, dataset.text)
-
-                    if y_proba is not None:
-                        ensure_predictions_proba(y_proba, y_pred)
-                        y_proba_dict = dict(zip(dataset.get_original_text_indexes(), y_proba))
-                        probas.update({dataset.name: y_proba_dict})
+        for dataset, y_pred, y_proba in (
+            (train, y_pred_train, y_proba_train),
+            (test, y_pred_test, y_proba_test),
+        ):
+            if dataset is None:
+                continue
 
+            if dataset.is_multi_label_classification():
+                _validate_multilabel(
+                    dataset=dataset,
+                    predictions=y_pred,
+                    probabilities=y_proba,
+                    n_of_classes=len(model_classes)
+                )
+                if y_pred is not None:
+                    y_pred = np.array(y_pred, dtype='float')
+                elif y_proba is not None:
+                    y_pred = (np.array(y_proba) > multilabel_proba_threshold)
+                    y_pred = [np.array(model_classes)[pred] for pred in y_pred]
```

Comment:
No that line should be deleted for multilabel, the y_pred should be a binary matrix

Reply:
I replaced it with 
```
y_pred = (np.array(y_proba) > multilabel_proba_threshold)
y_pred = y_pred.astype(int)
```